src/OSPSuite.Core/Domain/Formulas/DistributionFormulaFactory.cs

```diff
@@ -21,68 +25,85 @@ public interface IDistributionFormulaFactory
-         return distribution;
-      }
+         addObjectPathToDistributionFormula(distributionFormula, minParameterName, MINIMUM, dimension);
+         addObjectPathToDistributionFormula(distributionFormula, maxParameterName, MAXIMUM, dimension);
+      };
 
-      public UniformDistributionFormula CreateUniformDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter minParameter, IParameter maxParameter)
+      public DistributionFormula CreateFor(DistributionType distributionType, IDimension dimension)
       {
-         var distribution = createDistribution<UniformDistributionFormula>();
-         distribution.Dimension = minParameter.Dimension;
-         addObjectPathToDistributionFormula(distribution, distributedParameter,minParameter,Constants.Distribution.MINIMUM);
-         addObjectPathToDistributionFormula(distribution, distributedParameter,maxParameter,Constants.Distribution.MAXIMUM);
-         return distribution;
+         switch (distributionType)
+         {
+            case DistributionType.Normal:
+               return createDistribution<NormalDistributionFormula>(dimension, configureNormalDistribution());
+            case DistributionType.LogNormal:
+               return createDistribution<LogNormalDistributionFormula>(dimension, configureLogNormalDistributionDistribution());
+            case DistributionType.Uniform:
+               return createDistribution<UniformDistributionFormula>(dimension, configureLogNormalDistributionDistribution());
+            case DistributionType.Discrete:
+               return createDistribution<DiscreteDistributionFormula>(dimension, configureDiscreteDistribution());
+            default:
+               throw new ArgumentOutOfRangeException(nameof(distributionType), distributionType, null);
+         }
       }
 
-      private TDistribution createDistribution<TDistribution>() where TDistribution:IDistributionFormula
+      private TDistribution createDistribution<TDistribution>(IDimension dimension, Action<DistributionFormula, IDimension> configure) where TDistribution : DistributionFormula
       {
-         return _objectBaseFactory.CreateObjectBaseFrom<TDistribution>(typeof (TDistribution));
+         var distribution = _objectBaseFactory.CreateObjectBaseFrom<TDistribution>(typeof(TDistribution))
+            .WithDimension(dimension);
+
+         configure(distribution, dimension);
+         return distribution;
       }
 
-      private void addObjectPathToDistributionFormula(IDistributionFormula distribution,
-                                                      IDistributedParameter distributedParameter,
-                                                      IParameter usedParameter,
-                                                      string alias)
+      private void addObjectPathToDistributionFormula(DistributionFormula distribution, string usedParameter, string alias, IDimension dimension)
```

Comment:
looking at this function and all the configureXXXDistribution functions above:
* I would remove the `alias `argument here (and also rename `usedParameter `to `usedParameterName`)
* and change the signature of configurXXX so that the default alias is also the default value of the argument, e.g.
`configureNormalDistribution(string meanParameterName = MEAN, string deviationParameterName = DEVIATION)`

Reply:
alias need to be MEAN or DEVIATION etc. other wise methods won't work
    protected virtual double Mean(IUsingFormula refObject)
      {
         return GetReferencedEntityByAlias(Constants.Distribution.MEAN, refObject).Value;
      }

Reply:
all done